utils/puppeteerScrapper.js

```diff
@@ -1,151 +1,90 @@
-    console.info("Waiting the href");
-    await waitForSelectors([[hrefSelector]], targetPage, { timeout, visible: true });
-    console.info("Getting the href");
-    const [href] = await targetPage.$$eval(hrefSelector, elements => {
-      return elements.map(elem => {
-        return elem.getAttribute('href');
-      })
-    });
-
-    console.info("Waiting name of the waifu");
-    await waitForSelectors([[nameSelector]], targetPage, { timeout, visible: true });
-    console.info("Getting name of the waifu");
-    const [name] = await targetPage.$$eval(nameSelector, elements => {
-      return elements.map(elem => {
-        return elem.textContent;
-      })
-    });
-
-    console.info("Waiting image of the waifu");
-    await waitForSelectors(["#widget-waifu-of-the-day > a > img"], targetPage, { timeout, visible: true });
-    console.info("Getting image of the waifu");
-    const [image] = await targetPage.$$eval("#widget-waifu-of-the-day > a > img", elements => {
-      return elements.map(elem => {
-        return elem.getAttribute('src');
-      })
-    });
-
-    waifu.name = name;
-    waifu.bioUrl = `https://mywaifulist.moe${href}`;
-    waifu.imageUrl = image;
-    await Promise.all(promises);
-  }
-  {
-    const targetPage = page;
-    console.info("Waiting the image to be ready");
-    const element = await waitForSelectors([[imageSelector]], targetPage, { timeout, visible: true });
-    console.info("Clicking the image");
-    await element.click({
-      offset: {
-        x: 200,
-        y: 209,
-      },
-    });
-
-    console.info("Waiting the Description to be ready");
-    await waitForSelectors([[descriptionSelector]], targetPage, { timeout, visible: true });
-    console.info("Getting Description of the waifu");
-    const [extract] = await targetPage.$$eval(descriptionSelector, elements => {
-      return elements.map(elem => {
-        return elem.textContent;
-      })
-    });
-    waifu.extract = extract;
-
-    console.info("Waiting the anime where the waifu appears to be ready");
-    await waitForSelectors([[appearsSelector]], targetPage, { timeout, visible: true });
-    console.info("Getting the anime where the waifu appears");
-    const [appearsIn] = await targetPage.$$eval(appearsSelector, elements => {
-      return elements.map(elem => {
-        return {
-          url: `https://mywaifulist.moe${elem.getAttribute('href')}`,
-          text: elem.textContent
-        };
-      })
-    });
-
-    waifu.appearsIn = appearsIn;
+  const nameElement = await page.$(nameSelector)
+  const name = await page.evaluate(el => el.textContent, nameElement)
+  console.info("Name: OK...")
 
-  }
+  const imgElement = await page.$(`[alt="${name.trim()}"]`)
+  const imageUrl = await page.evaluate(el => el.src, imgElement)
+  console.info("Image: OK...")
 
-  console.info("Closing the browser");
-  await browser.close();
+  const whereElement = await page.$(appearsSelector)
```

Comment:
Change `whereElement` to more explicit name, i.e: `whereTheWaifuAppearsOnElement`